File: app/src/main/java/me/ykrank/s1next/data/api/model/ForumSearchResult.java
package me.ykrank.s1next.data.api.model;

/**
 * Created by ykrank on 2016/10/18.
 */

public class ForumSearchResult {

    private String content;


    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ForumSearchResult forumSearchResult = (ForumSearchResult) o;

        return content.equals(forumSearchResult.content);

    }

    @Override
    public int hashCode() {
        return content.hashCode();
    }

    @Override
    public String toString() {
        return "ForumSearchResult{" +
                "content='" + content + '\'' +
                '}';
    }
}

File: app/src/main/java/me/ykrank/s1next/data/api/model/search/ForumSearchResult.kt
package me.ykrank.s1next.data.api.model.search

import android.text.Spanned
import com.github.ykrank.androidtools.ui.adapter.StableIdModel
import me.ykrank.s1next.widget.span.HtmlCompat

/**
 * Created by ykrank on 2016/10/18.
 */

class ForumSearchResult : StableIdModel, SearchResult() {

    var content: String? = null
    private var _htmlContent: Spanned? = null
    val htmlContent: Spanned?
        get() = if (_htmlContent != null || content == null)
            _htmlContent
        else
            HtmlCompat.fromHtml(content, HtmlCompat.FROM_HTML_MODE_COMPACT_EXCLUDE_BLOCKQUOTE)

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val forumSearchResult = o as ForumSearchResult?

        return content == forumSearchResult!!.content

    }

    override fun hashCode(): Int {
        return content!!.hashCode()
    }

    override fun toString(): String {
        return "ForumSearchResult{" +
                "content='" + content + '\''.toString() +
                '}'.toString()
    }
}
